### PR TITLE
Relax validation for responses with empty content status.

### PR DESCRIPTION
**Pull request by minwoox**
Motivation:
An exception is currently raised when aggregating a response if the response's status indicates empty content but content is present. However, a third-party legacy server might send a response with content despite an empty content status, which is beyond the client's control.

Modification:
- Changed the behavior to log a debug message instead of raising an exception when content exists for a status that indicates empty content.

Result:
- Armeria client now works with legacy servers that send content with an empty content status without raising exceptions.

### CODE REVIEW

File: core/src/main/java/com/linecorp/armeria/common/HttpResponseWriter.java [hunk 1]
```diff
@@ -34,18 +34,18 @@ default void close(AggregatedHttpResponse res) {
...
             content = res.content();
-            final boolean contentAlwaysEmpty = isContentAlwaysEmptyWithValidation(status, content);
-
             if (!tryWrite(headers)) {
                 return;
             }
 
-            // Add content if not empty.
-            if (!contentAlwaysEmpty && !content.isEmpty()) {
+            if (headers.status().isContentAlwaysEmpty()) {
```
Comment (ikhoon) [inline core/src/main/java/com/linecorp/armeria/common/HttpResponseWriter.java:43]: Need to release the resource?
```suggestion
            if (headers.status().isContentAlwaysEmpty()) {
                content.close();
```

Comment (minwoox) [reply core/src/main/java/com/linecorp/armeria/common/HttpResponseWriter.java:43]: It's released here. 😉 
https://github.com/line/armeria/pull/5862/files/f91510580f795596cb30717b66b423c3d3af9ad1#diff-f288d29e01958b60c9befbce9e62c6aa2da2545dcc4f354047eb3970bd7816adR63

Comment (ikhoon) [reply core/src/main/java/com/linecorp/armeria/common/HttpResponseWriter.java:43]: I missed that point.

**Review by ikhoon — APPROVED** [review]
👍👍

File: core/src/main/java/com/linecorp/armeria/common/HttpResponseUtil.java [hunk 2]
```diff
@@ -20,10 +20,15 @@
...
 import java.util.concurrent.CompletableFuture;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import com.linecorp.armeria.common.util.Exceptions;
 
 final class HttpResponseUtil {
 
+    static final Logger logger = LoggerFactory.getLogger(HttpResponseUtil.class);
```
Comment (jrhee17) [inline core/src/main/java/com/linecorp/armeria/common/HttpResponseUtil.java:30]: optional) `HttpResponseUtil` seems like an package-private class which makes it susceptible to us refactoring this class without realizing that users are using this logger pattern

**Review by jrhee17 — APPROVED** [review]
👍 👍 👍

File: core/src/main/java/com/linecorp/armeria/common/HttpResponseUtil.java (hunk 2 above)
Comment (minwoox) [reply core/src/main/java/com/linecorp/armeria/common/HttpResponseUtil.java:30]: Add a comment for that. 😉

**Pull request merged**